src/main/java/uk/gov/justice/digital/job/DataHubCdcJob.java

```diff
@@ -0,0 +1,115 @@
+    private final TableStreamingQueryProvider tableStreamingQueryProvider;
+    private final TableDiscoveryService tableDiscoveryService;
+
+    @Inject
+    public DataHubCdcJob(
+            JobArguments arguments,
+            JobProperties properties,
+            SparkSessionProvider sparkSessionProvider,
+            TableStreamingQueryProvider tableStreamingQueryProvider,
+            TableDiscoveryService tableDiscoveryService) {
+        logger.info("Initializing DataHubCdcJob");
+        this.arguments = arguments;
+        this.properties = properties;
+        this.sparkSessionProvider = sparkSessionProvider;
+        this.tableStreamingQueryProvider = tableStreamingQueryProvider;
+        this.tableDiscoveryService = tableDiscoveryService;
+        logger.info("DataHubCdcJob initialization complete");
+    }
+
+    public static void main(String[] args) {
+        logger.info("Job started");
+        PicocliRunner.run(DataHubCdcJob.class, MicronautContext.withArgs(args));
+    }
+
+    @Override
+    public void run() {
+        boolean runLocal = System.getProperty(SPARK_JOB_NAME_PROPERTY) == null;
+        if(runLocal) {
+            logger.info("Running locally");
+            SparkConf sparkConf = new SparkConf().setAppName("DataHubCdcJob local").setMaster("local[*]");
+            SparkSession spark = sparkSessionProvider.getConfiguredSparkSession(sparkConf, arguments.getLogLevel());
+            runJob(spark);
+            waitUntilQueryTerminates(spark);
+        } else {
+            logger.info("Running in Glue");
+            String jobName = properties.getSparkJobName();
+            GlueContext glueContext = sparkSessionProvider.createGlueContext(jobName, arguments.getLogLevel());
+            SparkSession spark = glueContext.getSparkSession();
+            Job.init(jobName, glueContext, arguments.getConfig());
+            runJob(spark);
+            waitUntilQueryTerminates(spark);
+            Job.commit();
+        }
+    }
+
+    /**
+     * The main entry point for starting a streaming application to process all micro-batches continuously for all tables.
+     */
+    @VisibleForTesting
+    List<TableStreamingQuery> runJob(SparkSession spark) {
+        logger.info("Initialising Job");
+        List<ImmutablePair<String, String>> tablesToProcess = tableDiscoveryService.discoverTablesToProcess();
+        List<TableStreamingQuery> streamingQueries = new ArrayList<>();
+
+        if(!tablesToProcess.isEmpty()) {
+            tablesToProcess.forEach(tableDetails -> {
+                String inputSchemaName = tableDetails.getLeft();
+                String inputTableName = tableDetails.getRight();
+                TableStreamingQuery streamingQuery = tableStreamingQueryProvider.provide(inputSchemaName, inputTableName);
+                streamingQuery.runQuery(spark);
+                streamingQueries.add(streamingQuery);
+            });
+        } else {
+            logger.warn("No tables to process");
```

Comment:
I was unsure of whether to throw and crash out here instead but I figured you might conceivably legitimately deploy a streaming app for some set of tables and only later add the configs for the tables and restart it.

Reply:
I wonder if it may mask a misconfiguration. I guess we could verify from the logs if a table is being processed.